unitTests/ui/MLFacetManagerTest.ts

```diff
@@ -0,0 +1,134 @@
+    }
+
+    function queryFacetsResponse(): IFacetResponse[] {
+      return [MLFacetTestUtils.getCompleteFacetResponse(facets[1]), MLFacetTestUtils.getCompleteFacetResponse(facets[0])];
+    }
+
+    it('should disable the component if it contains no MLFacet child', () => {
+      test.env.searchInterface.getComponents = () => [];
+      triggerAfterComponentsInitialization();
+      expect(test.cmp.disabled).toBe(true);
+    });
+
+    it('should disable the component if a query response has no facets parameter', () => {
+      $$(test.env.root).trigger(QueryEvents.querySuccess, { results: {} });
+      expect(test.cmp.disabled).toBe(true);
+    });
+
+    it('should have the component in the right order', () => {
+      expect(managerContainerChildren()[0]).toBe(facets[0].element);
+      expect(managerContainerChildren()[1]).toBe(facets[1].element);
+    });
+
+    it('should not disable the component if a query response has no facets parameter', () => {
+      triggerAfterComponentsInitialization();
+      expect(test.cmp.disabled).toBe(false);
+    });
+
+    it('should reorder the facets correctly in the DOM depending on the query results', () => {
+      triggerAfterComponentsInitialization();
+      triggerQuerySuccess(queryFacetsResponse());
+
+      expect(managerContainerChildren()[0]).toBe(facets[1].element);
+      expect(managerContainerChildren()[1]).toBe(facets[0].element);
+    });
+
+    it('should remove the facets not in the response', () => {
+      triggerAfterComponentsInitialization();
+      triggerQuerySuccess([]);
+
+      expect(managerContainerChildren().length).toBe(0);
+    });
+
+    it(`when the "enableReorder" option is "false"
+    should not reorder the facets`, () => {
+      options = {
+        enableReorder: false
+      };
+      initializeComponent();
+      triggerAfterComponentsInitialization();
+      triggerQuerySuccess(queryFacetsResponse());
+
+      expect(managerContainerChildren()[0]).toBe(facets[0].element);
+      expect(managerContainerChildren()[1]).toBe(facets[1].element);
+    });
+
+    it(`when the "onUpdate" option is defined
+    should call it for every updated facets`, () => {
+      options = {
+        onUpdate: jasmine.createSpy('onUpdate')
+      };
+      initializeComponent();
+      triggerAfterComponentsInitialization();
+      triggerQuerySuccess(queryFacetsResponse());
+
+      expect(options.onUpdate).toHaveBeenCalledTimes(queryFacetsResponse().length);
+    });
+
+    it(`when the "compareFacets" option is defined
+    should use it to reorder facets`, () => {
+      options = {
+        compareFacets: (facetA, facetB) => {
+          return facetB.options.numberOfValues - facetA.options.numberOfValues;
+        }
+      };
+      initializeComponent();
+      triggerAfterComponentsInitialization();
+      triggerQuerySuccess(queryFacetsResponse());
+
+      expect(managerContainerChildren()[0]).toBe(facets[0].element);
```

Comment:
Would it be better to configure compareFacets to return the facets in a different order from the initial order? At the moment, it's not clear if the `compareFacets` function is being used.